app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,107 @@
+class MarketsController < ApplicationController
+
+  before_action :get_market, only: [:edit, :show, :update]
+
+  def get_market
+    @market = Market.find(params[:market_id] || params[:id])
+  end
+
+  def show
+    get_market
```

Comment:
Since this is in your `before_action` you don't need to also call this method here since it will be called due to the `before_action`